crowbar_framework/app/models/service_object.rb

```diff
@@ -925,6 +925,9 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     apply_locks = []
     applying_nodes = []
 
+    # Cache some node attributes to avoid useless node reloads
+    node_attr_cache = {}
```

Comment:
@aspiers ideally yes, but it's more intrusive and I wanted to optimize the apply_role case in a backportable way.